File: DissertationPhd/magnetic control/trajectory.py
# ---> в 00_tiny_functions.py

def trajectory(w: float, C: list, t:float) -> np.ndarray: 
    return np.array([-3*C[0]*w*t + 2*C[1]*np.cos(w*t) - 2*C[2]*np.sin(w*t) + C[3],
                     C[4]*np.sin(w*t) + C[5]*np.cos(w*t),
                     2*C[0] + C[1]*np.sin(w*t) + C[2]*np.cos(w*t),
                     -3*C[0]*w - 2*C[1]*w*np.sin(w*t) - 2*C[2]*w*np.cos(w*t),
                     C[4]*w*np.cos(w*t) - C[5]*w*np.sin(w*t),
                     C[1]*w*np.cos(w*t) - C[2]*w*np.sin(w*t)])

File: DissertationPhd/magnetic_control/trajectory.m
function [X]=trajectory(w,C,t) 
X(:,1)=-3*C(1)*w*t+2*C(2)*cos(w*t)-2*C(3)*sin(w*t)+C(4);
X(:,2)=C(5)*sin(w*t)+C(6)*cos(w*t);
X(:,3)=2*C(1)+C(2)*sin(w*t)+C(3)*cos(w*t);
X(:,4)=-3*C(1)*w-2*C(2)*w*sin(w*t)-2*C(3)*w*cos(w*t);
X(:,5)=C(5)*w*cos(w*t)-C(6)*w*sin(w*t);
X(:,6)=C(2)*w*cos(w*t)-C(3)*w*sin(w*t);
end
